OpenRA.Mods.Common/Scripting/Properties/GeneralProperties.cs

```diff
@@ -99,13 +99,15 @@ public class GeneralProperties : ScriptActorProperties
 	{
 		readonly IFacing facing;
 		readonly AutoTarget autotarget;
+		readonly Tooltip tooltip;
 		readonly ScriptTags scriptTags;
 
 		public GeneralProperties(ScriptContext context, Actor self)
 			: base(context, self)
 		{
 			facing = self.TraitOrDefault<IFacing>();
 			autotarget = self.TraitOrDefault<AutoTarget>();
+			tooltip = self.TraitOrDefault<Tooltip>();
```

Comment:
An actor can easily have multiple Tooltip traits (In RV IFV does). Use TraitsImplementing and use the first enabled one.

Reply:
Upstream examples are TS Tick Tank and Artillery.